tests/pkg_api/test_utils.py

```diff
@@ -206,28 +176,54 @@ def test_get_statement_representation(
         pkg_data_example: PKG data example.
         statement_representation: Expected statement representation.
     """
+    statement_node_id = utils.get_statement_node_id(pkg_data_example)
     assert strip_string(
-        utils._get_statement_representation(pkg_data_example, "_:st")
+        utils._get_statement_representation(pkg_data_example, statement_node_id)
     ) == strip_string(statement_representation)
 
 
 def test_get_query_for_add_statement(
     pkg_data_example: PKGData,
     statement_representation: str,
-    preference_representation: str,
 ) -> None:
     """Tests _get_query_for_add_statement method.
 
     Args:
         pkg_data_example: PKG data example.
-        statement_representation: Expected statement representation.
-        preference_representation: Expected preference representation.
+        statement_representation: Statement representation.
     """
     sparql_query = f"""INSERT DATA {{
         {statement_representation}
-
-        {preference_representation}
     }}"""
     assert utils.get_query_for_add_statement(pkg_data_example) == strip_string(
         sparql_query
     )
+
+
+def test_get_query_for_add_preference(
+    pkg_data_example: PKGData,
+) -> None:
+    """Tests _get_query_for_add_preference method.
+
+    Args:
+        pkg_data_example: PKG data example.
+    """
+    sparql_query = """
+        INSERT {
+            ?subject wi:preference [
+                pav:derivedFrom ?statement ;
+                wi:topic ?object ; wo:weight [
+                    wo:weight_value "-1.0"^^xsd:decimal;
+                    wo:scale pkg:StandardScale
+                ]
+            ] .
+        }
+        WHERE {
+            ex:abcac10b-58cc-4372-a567-0e02b2c3d479 a rdf:Statement ;
+            rdf:subject ?subject; rdf:object ?object .
+            BIND(ex:abcac10b-58cc-4372-a567-0e02b2c3d479 AS ?statement)
+        }
+    """
+    assert utils.get_query_for_add_preference(pkg_data_example) == strip_string(
+        sparql_query
+    )
```

Comment:
Test for get_query_for_conditioned_get_preference is missing

Reply:
Added corresponding test